sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/EventHubScope.cs

```diff
@@ -144,8 +217,68 @@ await Task.WhenAll
                     })
                 );
 
-                return new EventHubScope(eventHub.Name, groups);
+                return new EventHubScope(eventHub.Name, groups, wasEventHubCreated: true);
+            }
+        }
+
+        /// <summary>
+        ///   It creates an instance of <see cref="NamespaceProperties"/>, populates it from a connection string and returns it.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string from the existing Event Hubs namespace</param>
+        ///
+        /// <returns>The <see cref="NamespaceProperties" /> that will be used in a given test run.</returns>
+        ///
+        /// <exception cref="ArgumentException">Occurs when <param name="connectionString"/> holds an invalid connection string.</exception>
+        ///
+        public static NamespaceProperties PopulateNamespacePropertiesFromConnectionString(string connectionString)
+        {
+            if (IsConnectionStringValid(connectionString,
+                                        out string namespaceName))
+            {
+                return new NamespaceProperties(namespaceName, connectionString, wasNamespaceCreated: false);
+            }
+
+            throw new ArgumentException("An endpoint could not be found in the passed connection string");
+        }
+
+        /// <summary>
+        ///   It checks if the connection string contains a valid namespace name.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to be parsed.</param>
+        /// <param name="namespaceName">The namespace name taken from the connection string.</param>
+        ///
+        /// <returns>
+        ///   <c>true</c> if the connection string is a valid connection string; otherwise, <c>false</c>.
+        /// </returns>
+        ///
+        private static bool IsConnectionStringValid(string connectionString,
+                                                    out string namespaceName)
+        {
+            namespaceName = ParseNamespaceName(connectionString);
+
+            return !string.IsNullOrEmpty(namespaceName);
+        }
+
+        /// <summary>
+        ///   It returns the namespace name from a give set of connection string properties.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to be parsed.</param>
+        ///
+        /// <returns>The namespace name for a given connection string.</returns>
+        ///
+        private static string ParseNamespaceName(string connectionString)
```

Comment:
nit:  Would you be so kind as to reorder the methods to that the `public` ones come before the `private` ones?